src/QsCompiler/Transformations/ClassicallyControlled.cs

```diff
@@ -996,8 +996,16 @@ public override QsStatementKind OnConditionalStatement(QsConditionalStatement st
                         else
                         {
                             // Lift the scope to its own operation
-                            if (this.SharedState.LiftBody(block.Body, false, out var call, out var callable))
+                            if (this.SharedState.LiftBody(block.Body, null, false, out var call, out var callable))
                             {
+                                // Make sure the inferred information for the callable has local quantum dependency
+                                call = new TypedExpression(
+                                    call.Expression,
+                                    call.TypeArguments,
+                                    call.ResolvedType,
+                                    new InferredExpressionInformation(false, true),
```

Comment:
To make sure I understand, does adding the quantum dependency here apply to the call expression or to the whole lifted function? If the former then I think this makes sense, but I'm a little confused if it means the whole lifted function would inherit a dependency.

Reply:
This just sets the inferred information for the call expression, so it won't affect the lifted callable. The calls for the classically controlled are set this way to reflect the fact that these calls are dependent on the results of quantum measurements.